oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/BearerTokenAuthenticationToken.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.server.resource;
+
+import java.util.Collections;
+
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.SpringSecurityCoreVersion;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider;
+import org.springframework.security.oauth2.server.resource.web.BearerTokenAuthenticationFilter;
+import org.springframework.util.Assert;
+
+/**
+ * An {@link Authentication} that contains a
+ * <a href="https://tools.ietf.org/html/rfc6750#section-1.2" target="_blank">Bearer Token</a>.
+ *
+ * Used by {@link BearerTokenAuthenticationFilter} to prepare an authentication attempt and supported
+ * by {@link JwtAuthenticationProvider}.
+ *
+ * @author Josh Cummings
+ * @since 5.1
+ */
+public class BearerTokenAuthenticationToken extends AbstractAuthenticationToken {
+	private static final long serialVersionUID = SpringSecurityCoreVersion.SERIAL_VERSION_UID;
+
+	private String token;
+
+	/**
+	 * Create a {@code BearerTokenAuthenticationToken} using the provided parameter(s)
+	 *
+	 * @param token
+	 */
+	public BearerTokenAuthenticationToken(String token) {
+		super(Collections.emptyList());
+
+		Assert.hasText(token, "token cannot be empty");
+
+		this.token = token;
+	}
+
+	/**
+	 * Get the <a href="https://tools.ietf.org/html/rfc6750#section-1.2" target="_blank">Bearer Token</a>
+	 * @return
+	 */
+	public String getToken() {
+		return this.token;
+	}
+
+	/**
+	 * Get the <a href="https://tools.ietf.org/html/rfc6750#section-1.2" target="_blank">Bearer Token</a>
+	 * @return
+	 */
+	@Override
+	public Object getCredentials() {
+		return this.getToken();
+	}
+
+	/**
+	 * Get the <a href="https://tools.ietf.org/html/rfc6750#section-1.2" target="_blank">Bearer Token</a>
+	 * @return
+	 */
+	@Override
+	public Object getPrincipal() {
+		return this.getToken();
```

Comment:
I believe this comes down to whether or not the token is a principal at all. Note that `UsernamePasswordAuthenticationToken`, when used as an authentication request, does take the value from the constructor, set it as the principal, and returns it from getPrincipal. The difference is that `username` is more clearly a principal than `token`.

@rwinch and I were just discussing this one a few days ago. Another option is to return `null`. Really for me, the best `Authentication` would be one that is principal-less since Bearer Tokens are primarily about authority and incidentally about identity (say, when there is a `sub`).

Returning `null` clearly states that there is no "entity" that is part of this request, returning `null` just feels like a smell.

Thoughts, @rwinch ?

Reply:
I still struggle with this myself, but I think at least or now we should leave as is. Perhaps we should create a ticket to consider this separately.

Reply:
I personally feel `getCredentials()` should return the `token` and `getPrincipal()` should return `""`.
Logging a ticket to revisit is fine with me too.